core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -4169,7 +4177,8 @@ object ZIO extends ZIOCompanionPlatformSpecific {
 
   private[zio] final class Fork[R, E, A](
     val value: ZIO[R, E, A],
-    val scope: Option[ZScope[Exit[Any, Any]]]
+    val scope: Option[ZScope[Exit[Any, Any]]],
+    val reportFailure: Option[Cause[Any] => Unit]
```

Comment:
Adding an additional field here is a binary incompatible change even though this is package private. We could potentially revert and use the previous implementation of `forkInternal` for everything except the `raceWith` implementation.

Reply:
Safe to mark as an exclusion I think

Reply:
I think so too. Will do.